worlds/zillion/__init__.py

```diff
@@ -249,37 +253,30 @@ def create_items(self) -> None:
                 self.logger.debug(f"Zillion Items: {item_name}  1")
                 self.multiworld.itempool.append(self.create_item(item_name))
 
-    def set_rules(self) -> None:
-        # logic for this game is in create_regions
-        pass
-
+    @override
     def generate_basic(self) -> None:
         assert self.zz_system.randomizer, "generate_early hasn't been called"
         # main location name is an alias
-        main_loc_name = self.zz_system.randomizer.loc_name_2_pretty[self.zz_system.randomizer.locations['main'].name]
+        main_loc_name = self.zz_system.randomizer.loc_name_2_pretty[self.zz_system.randomizer.locations["main"].name]
 
         self.multiworld.get_location(main_loc_name, self.player)\
             .place_locked_item(self.create_item("Win"))
         self.multiworld.completion_condition[self.player] = \
             lambda state: state.has("Win", self.player)
 
     @staticmethod
-    def stage_generate_basic(multiworld: MultiWorld, *args: Any) -> None:
+    def stage_generate_basic(multiworld: MultiWorld, *args: Any) -> None:  # noqa: ANN401
         # item link pools are about to be created in main
         # JJ can't be an item link unless all the players share the same start_char
         # (The reason for this is that the JJ ZillionItem will have a different ZzItem depending
         #  on whether the start char is Apple or Champ, and the logic depends on that ZzItem.)
         for group in multiworld.groups.values():
-            # TODO: remove asserts on group when we can specify which members of TypedDict are optional
-            assert "game" in group
-            if group["game"] == "Zillion":
-                assert "item_pool" in group
+            if group["game"] == "Zillion" and "item_pool" in group:
                 item_pool = group["item_pool"]
                 to_stay: Chars = "JJ"
                 if "JJ" in item_pool:
-                    assert "players" in group
-                    group_players = group["players"]
-                    players_start_chars: List[Tuple[int, Chars]] = []
+                    group_players = set(group["players"])
```

Comment:
I has question: can a non-empty abstract set here be a frozenset?
If not, wouldn't an assert be better? You copy it and then assign it back, which seems fragile.
Alternatively `group["players"] = group_players = set(group["players"])` would be an option?

Reply:
Yeah, it did seem a little more fragile to separate those assignments.
I changed it to `group["players"] = group_players = set(group["players"])`

Even if the current code might not allow the `frozenset` with items, that seems more like an implementation detail that I shouldn't rely on.

I generated a few seeds with item links and looked at spoilers after this change.